inst/include/Rcpp/api/meat/DottedPairImpl.h

```diff
@@ -53,7 +53,7 @@ namespace Rcpp{
 		} else {
 			if( ref.isNULL( ) ) throw index_out_of_bounds() ;
 
-                        if( static_cast<R_xlen_t>(index) > ::Rf_xlength(ref.get__()) ) throw index_out_of_bounds() ;
+			if( static_cast<R_xlen_t>(index) > ::Rf_xlength(ref.get__()) ) throw index_out_of_bounds("index is out of bounds. Requested %i with %i available", static_cast<R_xlen_t>(index), ::Rf_xlength(ref.get__()) ) ;
```

Comment:
Minor, but to avoid overly long lines like this I often use an idiom like (pseudo-ish code):

```
if (condition) {
    const char* fmt =
        "Index out of bounds: [index=%s; extent=%s]"
    throw index_out_of_bounds(fmt, index, length);
}
```